static/src/js/components/GranuleResults/GranuleResultsItem.js

```diff
@@ -2,32 +2,179 @@
 import React from 'react'
 import { PropTypes } from 'prop-types'
 import { Waypoint } from 'react-waypoint'
+import queryString from 'query-string'
+import { Link } from 'react-router-dom'
+import { Dropdown } from 'react-bootstrap'
+
+import murmurhash3 from '../../util/murmurhash3'
 
 import './GranuleResultsItem.scss'
 
+class CustomDataLinksToggle extends React.Component {
+  constructor(props, context) {
+    super(props, context)
+    this.handleClick = this.handleClick.bind(this)
+  }
+
+  handleClick(e) {
+    e.preventDefault()
+    const { onClick } = this.props
+    onClick(e)
+  }
+
+  render() {
+    return (
+      <button
+        className="button granule-results-item__button"
+        type="button"
+        title="Download single granule data"
+        onClick={this.handleClick}
+      >
+        <i className="fa fa-download" />
+      </button>
+    )
+  }
+}
+
+CustomDataLinksToggle.propTypes = {
+  onClick: PropTypes.func.isRequired
+}
+
+const DataLinksButton = ({ dataLinks }) => {
+  if (dataLinks.length > 1) {
+    return (
+      <Dropdown>
+        <Dropdown.Toggle as={CustomDataLinksToggle} />
+        <Dropdown.Menu>
+          {
+            dataLinks.map((dataLink, i) => {
+              const key = `data_link_${i}`
+              return (
+                <Dropdown.Item key={key} href={dataLink.href} className="granule-results-item__dropdown-item">
+                  {dataLink.title}
+                </Dropdown.Item>
+              )
+            })
+          }
+        </Dropdown.Menu>
+      </Dropdown>
+    )
+  }
+  return (
+    <a
+      className="button granule-results-item__button"
+      href={dataLinks[0].href}
+      rel="noopener noreferrer"
+      title="Download single granule data"
+      target="_blank"
+    >
+      <i className="fa fa-download" />
+    </a>
+  )
+}
+
+DataLinksButton.propTypes = {
+  dataLinks: PropTypes.arrayOf(PropTypes.shape({})).isRequired
+}
+
 /**
  * Renders GranuleResultsItem.
  * @param {object} props - The props passed into the component.
  * @param {object} props.granule - Granule passed from redux store.
  */
-const GranuleResultsItem = ({ granule, isLast, waypointEnter }) => {
-  const title = granule.producer_granule_id ? granule.producer_granule_id : granule.title
-  const temporal = granule.formatted_temporal
+const GranuleResultsItem = ({
+  collectionId,
+  granule,
+  isLast,
+  location,
+  waypointEnter,
+  onExcludeGranule,
+  onFocusedGranuleChange
+}) => {
+  const handleRemoveClick = () => {
+    const {
+      id,
+      is_cwic: isCwic
+    } = granule
+
+    let hashedId
```

Comment:
Seems like this could be cleaned up by just overwriting `id` instead of creating a new variable and then `||` when we use it.

Reply:
Good call. Went ahead and made that change. @macrouch, just want to make sure you saw you had a chance to call out if it would be an issue for any reason.